### PR TITLE
[OPPRO-350] Remove backends-velox from backends-gazelle's dependency list

### PR DESCRIPTION
**Pull request by zhztheplayer**
To factor out some codes to remove the dependency from backends-velox to backend-gazelle. Common codes are moved to gluten-data.

Some classes shared by both two backends are renamed from "Velox... " to "Gluten... ".

Also renamed some utility classes with suffix "Utils" to "Util", since we are now using both in the codes.

### CODE REVIEW

File: backends-gazelle/pom.xml [hunk 1]
```diff
@@ -13,38 +13,17 @@
...
-                    <version>${project.version}</version>
-                    <scope>compile</scope>
-                </dependency>
-            </dependencies>
-        </profile>
-    </profiles>
+    <properties>
+        <maven.compiler.source>8</maven.compiler.source>
+        <maven.compiler.target>8</maven.compiler.target>
+        <resource.dir>${project.basedir}/src/main/resources</resource.dir>
```
Comment (jinchengchenghh) [inline backends-gazelle/pom.xml:19]: Why specify resource dir? The default dir is this dir

File: gluten-data/src/main/java/io/glutenproject/backendsapi/glutendata/GlutenDataIteratorApi.scala [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+import org.apache.spark.sql.connector.read.InputPartition
+import org.apache.spark.sql.execution.datasources.FilePartition
+import org.apache.spark.sql.execution.metric.SQLMetric
+import org.apache.spark.sql.utils.SparkArrowUtil
+import org.apache.spark.sql.utils.OASPackageBridge.InputMetricsWrapper
+import org.apache.spark.sql.vectorized.{ColumnarBatch, ColumnVector}
+import org.apache.spark.util.ExecutorManager
+import org.apache.spark.util.memory.TaskMemoryResources
+
+abstract class GlutenDataIteratorApi extends IIteratorApi with Logging {
```
Comment (jinchengchenghh) [inline gluten-data/src/main/java/io/glutenproject/backendsapi/glutendata/GlutenDataIteratorApi.scala:55]: GlutenDataIteratorApi is a bit long, how about GlutenIteratorApi?

Comment (zhztheplayer) [reply gluten-data/src/main/java/io/glutenproject/backendsapi/glutendata/GlutenDataIteratorApi.scala:55]: I am OK to both but as some extra information, GlutenDataIteratorApi is in module `gluten-data`. `GlutenXXX` is more suitable for contents under `gluten-core` to me?

File: backends-gazelle/pom.xml (hunk 1 above)
Comment (zhztheplayer) [reply backends-gazelle/pom.xml:19]: It's for targeting libgazelle_cpp.so.

Currently backends-gazelle reuse this setting from backends-velox.

Comment (zhztheplayer) [reply backends-gazelle/pom.xml:19]: Could you please help find some references about this?

File: gluten-data/src/main/java/io/glutenproject/backendsapi/glutendata/GlutenDataIteratorApi.scala (hunk 2 above)
Comment (zhztheplayer) [reply gluten-data/src/main/java/io/glutenproject/backendsapi/glutendata/GlutenDataIteratorApi.scala:55]: Agreed. I'll update the PR.

File: backends-gazelle/pom.xml (hunk 1 above)
Comment (jinchengchenghh) [reply backends-gazelle/pom.xml:19]: https://maven.apache.org/plugins/maven-resources-plugin/examples/resource-directory.html
I think we can remove all the resource.dir

Comment (zhztheplayer) [reply backends-gazelle/pom.xml:19]: I think I can raise another patch to remove all the `resource.dir` entries later on. Is that OK to you?

**Review by jinchengchenghh — APPROVED** [review]
wait CI

**Pull request merged**